data-prepper-plugins/otlp-source/src/main/java/org/opensearch/dataprepper/plugins/source/otlp/OTLPSource.java

```diff
@@ -0,0 +1,313 @@
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+
+@DataPrepperPlugin(name = "otlp", pluginType = Source.class, pluginConfigurationType = OTLPSourceConfig.class)
+public class OTLPSource implements Source<Record<Object>> {
+  private static final Logger LOG = LoggerFactory.getLogger(OTLPSource.class);
+  private static final String HTTP_HEALTH_CHECK_PATH = "/health";
+  private static final String REGEX_HEALTH = "regex:^/(?!health$).*$";
+  private static final String PIPELINE_NAME_PLACEHOLDER = "${pipelineName}";
+  static final String SERVER_CONNECTIONS = "serverConnections";
+
+  // Default RetryInfo with minimum 100ms and maximum 2s
+  private static final RetryInfoConfig DEFAULT_RETRY_INFO = new RetryInfoConfig(Duration.ofMillis(100),
+      Duration.ofMillis(2000));
+  private final OTLPSourceConfig otlpSourceConfig;
+  private final String pipelineName;
+  private final PluginMetrics pluginMetrics;
+  private final GrpcAuthenticationProvider authenticationProvider;
+  private final CertificateProviderFactory certificateProviderFactory;
+  private Server server;
+
+  @DataPrepperPluginConstructor
+  public OTLPSource(final OTLPSourceConfig otlpSourceConfig,
+      final PluginMetrics pluginMetrics,
+      final PluginFactory pluginFactory,
+      final PipelineDescription pipelineDescription) {
+    this(otlpSourceConfig, pluginMetrics, pluginFactory,
+        new CertificateProviderFactory(otlpSourceConfig), pipelineDescription);
+  }
+
+  // accessible only in the same package for unit test
+  OTLPSource(final OTLPSourceConfig otlpSourceConfig,
+      final PluginMetrics pluginMetrics,
+      final PluginFactory pluginFactory,
+      final CertificateProviderFactory certificateProviderFactory,
+      final PipelineDescription pipelineDescription) {
+    otlpSourceConfig.validateAndInitializeCertAndKeyFileInS3();
+    this.otlpSourceConfig = otlpSourceConfig;
+    this.pluginMetrics = pluginMetrics;
+    this.certificateProviderFactory = certificateProviderFactory;
+    this.pipelineName = pipelineDescription.getPipelineName();
+    this.authenticationProvider = createAuthenticationProvider(pluginFactory);
+  }
+
+  @Override
+  public void start(Buffer<Record<Object>> buffer) {
+    if (buffer == null) {
+      throw new IllegalStateException("Buffer provided is null");
+    }
+
+    if (server == null) {
+      @SuppressWarnings("unchecked")
+      Buffer<Record<? extends Metric>> metricBuffer = (Buffer<Record<? extends Metric>>) (Object) buffer;
+
+      final OTelLogsGrpcService oTelLogsGrpcService = new OTelLogsGrpcService(
+          (int) (otlpSourceConfig.getRequestTimeoutInMillis() * 0.8),
+          new OTelProtoStandardCodec.OTelProtoDecoder(),
```

Comment:
Why is the codec not configurable? What if I need the old data format?

Reply:
My team is currently building on the old data format. We have a lot of instances (>1000) running on it and would be blocked from your change until we managed to migrate our data format. Additionally, we are trying to implement OTLP/HTTP support in #5322. This would be easier to be done on just your change. But if we cannot use it with the old data format, we would need to implement it in the old sources as well. Therefore, I would highly appreciate support for both formats in the new otel-source as well. So far the old format has not been declared as deprecated AFAIK.